src/Server/TCPHandler.cpp

```diff
@@ -110,7 +110,10 @@ void TCPHandler::runImpl()
     setThreadName("TCPHandler");
     ThreadStatus thread_status;
 
-    session = std::make_unique<Session>(server.context(), ClientInfo::Interface::TCP, socket().secure());
+    /// For interserver mode we will create session later
+    if (!is_interserver_mode)
```

Comment:
How you can know here whether this is the interserver mode or not?
`receiveHello()` determines that but you haven't called it yet at this point!

Reply:
Thanks, I didn't noticed it, my bad. Currently it works only because `is_interserver_mode` is false by default.

The problem is that in interserver mode we do not really use this `session` object and just replace it with an object created later. And this object prints garbage in logs in its dtor (like in https://github.com/ClickHouse/ClickHouse/pull/35763#discussion_r869143673).